File: packages/shared/src/types/geometry.ts
/**
 * Core geometry types used across the application
 */

export interface Point {
  x: number
  y: number
}

export interface BoundingBox {
  minX: number
  minY: number
  maxX: number
  maxY: number
}

export interface TileCoord {
  row: number
  col: number
}

File: packages/shared/python/city_shared/geometry.py
"""Core geometry schemas used across the application."""

from pydantic import BaseModel


class Point(BaseModel):
    """A 2D point."""

    x: float
    y: float


class BoundingBox(BaseModel):
    """An axis-aligned bounding box."""

    min_x: float
    min_y: float
    max_x: float
    max_y: float


class TileCoord(BaseModel):
    """A tile coordinate in the grid."""

    row: int
    col: int
